File: leetcode/Offer05.py
class Solution:

    def replaceSpace(self, s: str) -> str:
        return '%20'.join(s.split(' '))

    def replaceSpace_(self, s: str) -> str:
        rst = ''
        i = 0
        for j in range(len(s)):
            if (s[j] == ' '):
                rst += s[i:j] + '%20'
                i = j + 1
        rst += s[i:]
        return rst

File: leetcode/Offer05.cpp
#include <string>

using namespace std;

class Solution {
public:
  string replaceSpace(string s) {
    string rst;
    int n = s.size();
    int l = 0;
    for (int i = 0; i < n; i++) {
      if (s[i] == ' ') {
        rst += s.substr(l, i - l) + "%20";
        l = i + 1;
      }
    }
    rst += s.substr(l);
    return rst;
  }
};